### PR TITLE
DX: lock SCA tools for PR builds

### PR DESCRIPTION
**Pull request by keradus**
We recently had an issue that master started to fail because on an update of PHPStan, starting detecting new issues.
To avoid confusion for contributors, we suggested on maintainers meeting to lock the dev-tools and updated them occasionally (eg once per Q). Same time, for master build, that is observed mostly by maintainers, we can still look for bleeding-edge version of SCA tools

### CODE REVIEW

**Comment by keradus** [comment]
cc @localheinz @julienfalque @kubawerlos @SpacePossum

File: dev-tools/composer.lock [hunk 1]
```diff
@@ -0,0 +1,4615 @@
...
+    "prefer-stable": false,
+    "prefer-lowest": false,
+    "platform": {
+        "php": "^7.4 || ^8.0"
+    },
+    "platform-dev": [],
+    "platform-overrides": {
+        "php": "7.4"
+    },
+    "plugin-api-version": "2.1.0"
```
Comment (kubawerlos) [inline dev-tools/composer.lock]: Update your Composer :)

**Review by kubawerlos — COMMENTED** [review]
I'd vote for removing `^` from `dev-tools/composer.json` instead of  having `dev-tools/composer.lock` committed, 4k lines of unnecessary noise.

**Comment by keradus** [comment]
> I'd vote for removing ^ from dev-tools/composer.json instead of having dev-tools/composer.lock committed

it won't lock the deep-deps (dependencies of dependency)

**Review by julienfalque — APPROVED** [review]

**Review by localheinz — APPROVED** [review]
👍

**Comment by kubawerlos** [comment]
> it won't lock the deep-deps (dependencies of dependency)

Obviously, but why do we need to lock them? If some dependency needs to be locked that mean it should be explicitly added to `composer.json`.

**Comment by keradus** [comment]
I wouldn't share that claim, myself.
You don't know which dependency needs to be locked until you explicitly experience issue because of it (like first-time contributor creates a PR and it failed on SCA for nothing related to given PR). Locking all SCA deps with lock-file prevents that, locking only 1st level dependencies (not using lock file) will only minimise the risk

File: dev-tools/composer.lock (hunk 1 above)
Comment (kubawerlos) [reply dev-tools/composer.lock:4611]: @keradus ping ☝🏼

Comment (kubawerlos) [reply dev-tools/composer.lock:4611]: @keradus ping ☝🏼

Comment (keradus) [reply dev-tools/composer.lock:4611]: i don't see any value in analysing the lock file on this moment and it doesn't bother anyone but you

Comment (kubawerlos) [reply dev-tools/composer.lock:4611]: > it doesn't bother anyone but you

I thought you were better than this... In fact I was hoping you would update your Composer and discover by yourself [this](https://getcomposer.org/doc/06-config.md#allow-plugins) new option of Composer 2.2 and update `composer.json` accordingly.

Comment (keradus) [reply dev-tools/composer.lock:4611]: you quoted part of my line. let me rephrase - I'm not aware of benefits, you didn't bother to mention any initially, and everyone else already approved the PR.

You want to be picky? tell ppl what to update.
You want to share some interesting idea, give the reasoning from start - it works way better ;)

Comment (kubawerlos) [reply dev-tools/composer.lock:4611]: >  tell ppl what to update.

I did :trollface: 

> Update your Composer :)

Well, it will stop working in July, so I guess no need to update it right now :P

**Comment by SpacePossum** [comment]
Thank you @keradus.
